Add to content_view_version versions asynchronical with a poll interval

tuxonfire opened the pull request:
patch to be able to Publish content_view versions asynchronical with a poll interval via the variables:

content_view_async
content_view_poll

tuxonfire committed:
Add to content_view_version versions asynchronical with a poll interval 

patch to be able to Publish content_view versions asynchronical with a poll interval via the variables:

content_view_async
content_view_poll

evgeni commented on roles/content_view_publish/tasks/main.yml:
```diff
@@ -13,6 +13,8 @@
     force_promote: "{{ content_view.force_promote | default(omit) }}"
     force_yum_metadata_regeneration: "{{ content_view.force_yum_metadata_regeneration | default(omit) }}"
     current_lifecycle_environment: "{{ content_view.current_lifecycle_environment | default(omit) }}"
+    async: "{{ content_view_async | default(omit) }}"
+    poll: "{{ content_view_poll | default(omit) }}"
```
These variables should be called `foreman_content_view_publish_async` and `foreman_content_view_publish_poll`

evgeni reviewed (changes requested):
This will require documenting the new variables in the [README.md](https://github.com/theforeman/foreman-ansible-modules/blob/develop/roles/content_view_publish/README.md) and adding changelog snippet (see https://github.com/theforeman/foreman-ansible-modules/blob/39ee4c4585ecd0109811de5b9183a49b4862aac6/changelogs/fragments/1591-cv-promote-parameters.yml for an example).

Shout if you need help with any of those.

tuxonfire committed:
Update main.yml

update of variables foreman_content_view_publish_async and foreman_content_view_publish_poll

tuxonfire committed:
Create 1676-content_view_version-async-poll.yml

content_view_version role adding foreman_content_view_publish_async and foreman_content_view_publish_poll

tuxonfire committed:
Update README.md

Adding optional variables foreman_content_view_publish_async and foreman_content_view_publish_poll

evgeni commented on changelogs/fragments/1676-content_view_version-async-poll.yml:
```diff
@@ -0,0 +1,4 @@
+minor_changes:
+  - content_view_version role adding foreman_content_view_publish_async and foreman_content_view_publish_poll (https://github.com/theforeman/foreman-ansible-modules/pull/1676)
+
+
```
```suggestion
```

evgeni commented on roles/content_view_publish/README.md:
```diff
@@ -12,6 +12,11 @@ This role supports the [Common Role Variables](https://github.com/theforeman/for
 
+### Optional  
+
+- `foreman_content_view_publish_async`: Asynchronous mode lets you control how long-running tasks execute.The default is 5 seconds. This can be too low on slower systems, or systems under heavy load.
```
```suggestion
- `foreman_content_view_publish_async`: Asynchronous mode lets you control how long-running tasks execute. See the [Ansible documentation](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_async.html#asynchronous-playbook-tasks) for details.
```

I wouldn't mention any defaults, they can change and are out of our control.

evgeni commented on roles/content_view_publish/README.md:
```diff
@@ -12,6 +12,11 @@ This role supports the [Common Role Variables](https://github.com/theforeman/for
+### Optional  
+
+- `foreman_content_view_publish_async`: Asynchronous mode lets you control how long-running tasks execute.The default is 5 seconds. This can be too low on slower systems, or systems under heavy load.
+- `foreman_content_view_publish_poll`:  For asynchronous tasks in this is how often to check back on the status of those tasks when an explicit poll interval is not supplied. The default is a reasonably moderate 15 seconds which is a tradeoff between checking in frequently and providing a quick turnaround when something may have completed.
```
```suggestion
- `foreman_content_view_publish_poll`: For asynchronous tasks, this is how often to check back on the status of those tasks.
```

Same as above, plus the "when an explicit poll interval is not supplied" doesn't apply here, as this very line supplies the explicit interval :)

evgeni reviewed (approved):
ACK on the change.

I feel free to apply the suggestions yourself, or I'll do it when merging.

evgeni commented on changelogs/fragments/1676-content_view_version-async-poll.yml:
```diff
@@ -0,0 +1,4 @@
+minor_changes:
+  - content_view_version role adding foreman_content_view_publish_async and foreman_content_view_publish_poll (https://github.com/theforeman/foreman-ansible-modules/pull/1676)
```
```suggestion
  - content_view_publish role - allow passing ``async`` and ``poll`` to the module (https://github.com/theforeman/foreman-ansible-modules/pull/1676)
```

evgeni committed:
Apply suggestions from code review
